File: src/game/components/snare.lua
local constants = require 'game.components.constants'

local Snare = {
  _meta = constants.Snare
}

function Snare.new(id)
  return {
    _meta = Snare._meta,
    id = id,
    strength = 1 -- number between 0 and 1
  }
end

return Snare

File: packages/client/src/game/components/Snare.ts
import { ComponentFlag } from '../flags'

export class Snare {
  static _id = 'Snare'
  _id = Snare._id

  static _flag = ComponentFlag.Snare
  _flag = ComponentFlag.Snare

  strength: number

  constructor(strength: number = 0) {
    this.strength = strength
  }
}
